procbuild/builder.py

```diff
@@ -80,7 +80,172 @@ def checkout(repo, branch, build_path):
     return shell('git clone %s --branch %s --single-branch %s' %
                  (repo, branch, build_path), retry=4)
 
+class BuildError(Exception):
+    
+    def __init__(self, message):
+        self.message = f"A build error occurred during the self.{message} step."
+
+class BuildManager(object):
+    
+    def __init__(self, 
```

Comment:
odd formatting here

Reply:
it makes it easier to change the variables without introducing confusing diffs since I didn't expect these to stay constant. I actually like this format if we end up having a lot of keyword arguments (which I think a lot of these are going to become), for similar reasons.

I'm also thinking that we might want to introduce a "type" system where we create an object first and then can check that everything is well formatted and pass in that instance to the BuildManager.

Reply:
The simpler we can keep things the better.  The words "type system" fill me with fear!

Reply:
i actually was putting quotes only around "type" to distinguish it from "type system". Basically I want one thing that does all the preamble to defining what the builder will need to know, instantiate that object (and make it debugable) and then pass that into the `BuildManager` which will do the actual build management. Right now the `BuildManager` is playing both roles.